### PR TITLE
Ea/story 13

### PR DESCRIPTION
**Pull request by eamouzou**
- Create test, controller, and views for login functionality

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -4,15 +4,27 @@ def new
...
+    if user.default?
+      redirect_to '/profile'
+    elsif user.merchant?
       redirect_to '/merchant/dashboard'
     elsif user.admin?
       redirect_to '/admin/dashboard'
     end
+    session[:user_id] = user.id
+    flash[:success] = "Welcome, #{user.name}!"
   end
```
Comment (eamouzou) [inline app/controllers/sessions_controller.rb:27]: This helper method allows us to redirect to the appropriate page based on the type of user.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@ def create
 
   def show
     require_user
+    if session[:user_id]
+      @user = User.find(session[:user_id]) 
+      flash[:success] = "#{@user.name} is logged in."
+    end
```
Comment (eamouzou) [inline app/controllers/users_controller.rb:25]: The conditional is just to make sure that there are no errors and no further process if there is no user.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h2><%= @user.name %>'s Profile</h2>
+Name: <%= @user.name %></br>
+Street Address: <%= @user.street_address %></br>
+City: <%= @user.city %></br>
+State: <%= @user.state %></br>
+Zip Code: <%= @user.zip %></br>
+Email: <%= @user.email %></br>
```
Comment (eamouzou) [inline app/views/users/show.html.erb:7]: The view for the user's show page to display their information.

File: spec/features/users/login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe "different users login" do
+  before(:each) do
+    @visitor_user = User.create!(name: "Damon",street_address: "2417 E. Curtis St. Denver, CO",
+                              city: "denver",state: "CO",zip: "80206",email: "new_email0@gmail.com",password: "hamburger0", role: 0)
```
Comment (eamouzou) [inline spec/features/users/login_spec.rb:6]: still need to add the test for the visitor user's experience as an edge case
